File: baekjoon/cpp/15651.cpp
#include <iostream>
using namespace std;

int N, M;
int arr[8]{};

void dfs(int level) {
    
    if(level == M) {
        for(int i = 0; i < M; i++) {
            cout << arr[i] << " ";
        }
        cout << "\n";
        return;
    }
    
    for(int i = 1; i <= N; i++) {
        arr[level] = i; 
        dfs(level + 1);
    }
    
    
}

int main()
{
    cin >> N >> M;
    dfs(0);
}

File: baekjoon/15651.py
n, m = map(int, input().split())
arr = [0] * m

def dfs(level):
    if level == m:
        print(*arr)
        return 

    for i in range(1, n+1):
        arr[level] = i
        dfs(level+1)


dfs(0)